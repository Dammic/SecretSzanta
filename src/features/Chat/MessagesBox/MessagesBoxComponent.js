'use strict'
import React from 'react'

const MessagesBoxComponent = ({
    messages = []
}) => {
    const buildMessage = (message) => {
        const {time, author, content, messageIndex} = message
        return (
            <div key={messageIndex} className="message">
                [{time}] {author === 'SERVER' ? '' : author}: {content}
            </div>
        )
    }

    return (
        <div className="messages-box">
            {messages.map((message) => buildMessage(message))}
        </div>
    )
}

export default MessagesBoxComponent


