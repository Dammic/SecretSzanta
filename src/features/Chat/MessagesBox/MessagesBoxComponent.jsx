import React from 'react'
import PropTypes from 'prop-types'
import { map } from 'lodash'
import classNames from 'classnames/bind'

import styles from './MessagesBox.css'

const buildMessage = (message, index) => {
    const { time, author, content } = message
    const displayedMessage = `[${time}]${author ? ` ${author}:` : ':'} ${content}`

    return (
        <div key={index} className={classNames(styles.message, { [styles.serverMessage]: !author })}>
            {displayedMessage}
        </div>
    )
}

const MessagesBoxComponent = React.forwardRef(({
    messages = [],
}, messagesBoxRef) => (
    <div className={styles.messagesBox} ref={messagesBoxRef}>
        {map(messages, (message, index) => buildMessage(message, index))}
    </div>
))

MessagesBoxComponent.displayName = 'MessagesBoxComponent'
MessagesBoxComponent.propTypes = {
    messages: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
}
export default MessagesBoxComponent

