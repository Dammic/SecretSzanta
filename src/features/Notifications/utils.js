import React from 'react'
import { compact, trim, map, fromPairs } from 'lodash'
import { Counter } from './components/Counter'

export const markupMessage = (template, values = {}) => {
    const regex = /({.*?})/g
    const matchedKeys = template.match(regex) || []
    const splitedNodes = compact(template.split(regex))

    const result = splitedNodes.map((val) => {
        if (matchedKeys.includes(val)) {
            const keyName = trim(val, ' {}')
            return values[keyName] || '[STEVE! YOU SHOULD put some value HERE]'
        }
        return val
    })

    return result
}

export const replaceKeysWithComponent = (values) => {
    return fromPairs(map(values, (val, key) => {
        const lowercasedKey = key.toLowerCase()
        if (lowercasedKey.includes('bold')) {
            return [key, <b key={key}>{val}</b>]
        }
        if (lowercasedKey.includes('counter')) {
            return [key, <Counter key={key} start={val} />]
        }
        return [key, val]
    }))
}
