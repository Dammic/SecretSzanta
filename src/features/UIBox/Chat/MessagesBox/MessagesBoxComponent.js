'use strict'
import React from 'react'
import classNames from 'classnames/bind'

const MessagesBoxComponent = ({
    messages = [],
    setMessagesEndRef
}) => {
    const buildMessage = (message, index) => {
        const {time, author, content} = message
        return (
            <div key={index} className={classNames("message", {'server-message': !author})}>
                [{time}]{author ? ` ${author}: ` : ':'} {content}
            </div>
        )
    }

    return (
        <div className="messages-box">
            {messages.map((message, index) => buildMessage(message, index))}
            <div id="EndOfMessages" ref={setMessagesEndRef} />
        </div>
    )
}

export default MessagesBoxComponent

