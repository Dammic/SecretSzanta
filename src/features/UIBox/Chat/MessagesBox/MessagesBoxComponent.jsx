import React from 'react'
import PropTypes from 'prop-types'
import { map } from 'lodash'
import classNames from 'classnames/bind'

import styles from './MessagesBox.css'

const buildMessage = (message, index) => {
    const { time, author, content } = message
    const displayedMessage = `[${time}]${author ? ` ${author}: ` : ':'} ${content}`

    return (
        <div key={index} className={classNames(styles.message, { [styles.serverMessage]: !author })}>
            {displayedMessage}
        </div>
    )
}

const MessagesBoxComponent = ({
    messages = [],
    setMessagesBoxRef,
}) => (
    <div className={styles.messagesBox} ref={setMessagesBoxRef}>
        {map(messages, (message, index) => buildMessage(message, index))}
    </div>
)

MessagesBoxComponent.propTypes = {
    messages: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
    setMessagesBoxRef: PropTypes.func,
}
export default MessagesBoxComponent

