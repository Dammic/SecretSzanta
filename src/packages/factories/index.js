export { arrayOf } from './arrayOf'
export { aMessage } from './messages'
export { aRoom, aRoomsList, aPlayer, aPlayersList } from './lobby'
