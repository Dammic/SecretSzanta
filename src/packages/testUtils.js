import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
