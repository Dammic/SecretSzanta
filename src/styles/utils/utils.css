@keyframes breathing {
  0% {
    transform: scale(0.8);
  }

  25% {
    transform: scale(0.9);
  }

  60% {
    transform: scale(1);
  }

  80% {
    transform: scale(0.9);
  }

  100% {
    transform: scale(0.8);
  }
}

