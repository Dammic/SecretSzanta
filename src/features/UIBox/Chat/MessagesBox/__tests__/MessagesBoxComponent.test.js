import React from 'react'
import { shallow } from 'enzyme'
import { expectMatchingSnapshot } from '../../../../../utils/testUtils'
import MessagesBoxComponent from '../MessagesBoxComponent'
import { aMessage, arrayOf } from '../../../../../utils/factories'

const setupProps = (propsOverrides = {}, renderMethod = shallow) => {
    const props = {
        setMessagesBoxRef: jest.fn(),
        messages: arrayOf(aMessage, 3, { seed: 1 }),
        ...propsOverrides,
    }
    const component = renderMethod(<MessagesBoxComponent { ...props } />)
    return { props, component }
}

describe('<MessagesBoxComponent />', () => {
    it('renders 3 messages if there are any passed', () => {
        const props = setupProps()
        expectMatchingSnapshot(<MessagesBoxComponent {...props} />)
    })

    it('renders no messages if no messages have been passed', () => {
        const props = setupProps({ messages: null })
        expectMatchingSnapshot(<MessagesBoxComponent {...props} />)
    })

    it('calls setMessagesBoxRef function on render', () => {
        const { props } = setupProps()
        expect(props.setMessagesBoxRef).toHaveBeenCalled()
    })
})
