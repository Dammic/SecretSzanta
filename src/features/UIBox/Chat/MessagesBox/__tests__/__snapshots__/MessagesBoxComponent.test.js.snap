// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessagesBoxComponent /> renders 3 messages if there are any passed 1`] = `
<MessagesBoxComponent
  component={
    <div
      className="messagesBox"
    >
      <div
        className="message"
        key="0"
      >
        [02:16] Zion.Reichel12:  Liaison
      </div>
      <div
        className="message"
        key="1"
      >
        [06:36] Audrey38:  Yen Operations
      </div>
      <div
        className="message"
        key="2"
      >
        [15:58] Heather_Hahn20:  SSL Beauty
      </div>
    </div>
  }
  props={
    Object {
      "messages": Array [
        Object {
          "author": "Zion.Reichel12",
          "content": "Liaison",
          "time": "02:16",
        },
        Object {
          "author": "Audrey38",
          "content": "Yen Operations",
          "time": "06:36",
        },
        Object {
          "author": "Heather_Hahn20",
          "content": "SSL Beauty",
          "time": "15:58",
        },
      ],
      "setMessagesBoxRef": [MockFunction],
    }
  }
>
  <div
    className="messagesBox"
  />
</MessagesBoxComponent>
`;

exports[`<MessagesBoxComponent /> renders no messages if no messages have been passed 1`] = `
<MessagesBoxComponent
  component={
    <div
      className="messagesBox"
    />
  }
  props={
    Object {
      "messages": null,
      "setMessagesBoxRef": [MockFunction],
    }
  }
>
  <div
    className="messagesBox"
  />
</MessagesBoxComponent>
`;
